### PR TITLE
add email validation - regex

### PR DESCRIPTION
**Pull request by rpalcolea**
Added email validation using regex.

While this could be accomplished with apache commons, two things:

* means adding a dependency
* it uses valid TLD domains that could potentially change and will make the plugin obsolete there

### CODE REVIEW

File: src/test/groovy/nebula/plugin/contacts/ContactsExtensionSpec.groovy [hunk 1]
```diff
@@ -77,4 +77,19 @@ class ContactsExtensionSpec extends Specification {
         mickey.github == 'mmouse'
         mickey.twitter == 'mmouse1928'
     }
+
+    def 'fails with invalid email'() {
```
Comment (sghill) [inline src/test/groovy/nebula/plugin/contacts/ContactsExtensionSpec.groovy]: @rpalcolea how would you feel about adding a test case for the happy path?

Comment (rpalcolea) [reply src/test/groovy/nebula/plugin/contacts/ContactsExtensionSpec.groovy]: Sure, will get back to you soon!

**Comment by rpalcolea** [comment]
Hi @sghill ,

Sorry for the delay.

Added the tests for happy path and also expanded a bit on scenarios for both valid and invalid emails.

**Comment by sghill** [comment]
Thanks @rpalcolea!

**Pull request merged**